src/RootCloseWrapper.js

```diff
@@ -18,31 +13,32 @@ function isModifiedEvent(event) {
   return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
 }
 
-function getSuppressRootClose() {
-  let id = CLICK_WAS_INSIDE + '_' + counter++;
-  return {
-    id,
-    suppressRootClose(event) {
-      // Tag the native event to prevent the root close logic on document click.
-      // This seems safer than using event.nativeEvent.stopImmediatePropagation(),
-      // which is only supported in IE >= 9.
-      event.nativeEvent[id] = true;
-    }
-  };
-}
-
 export default class RootCloseWrapper extends React.Component {
   constructor(props) {
     super(props);
 
     this.handleDocumentMouse = this.handleDocumentMouse.bind(this);
     this.handleDocumentKeyUp = this.handleDocumentKeyUp.bind(this);
+  }
 
-    let { id, suppressRootClose } = getSuppressRootClose();
+  componentDidMount() {
+    if (!this.props.disabled) {
+      this.bindRootCloseHandlers();
+    }
+  }
 
-    this._suppressRootId = id;
+  componentDidUpdate(prevProps) {
```

Comment:
generally cDU. this is fine, i don't think there is any reason why a child component would need the handlers registered before it rendered/updated.